pkg/cloudprovider/fake/cloudprovider.go

```diff
@@ -37,8 +38,25 @@ func (c *CloudProvider) Create(_ context.Context, constraints *v1alpha5.Constrai
 	for i := 0; i < quantity; i++ {
 		name := strings.ToLower(randomdata.SillyName())
 		instance := instanceTypes[0]
-		zone := instance.Zones().Intersection(constraints.Requirements.Zones()).UnsortedList()[0]
-		operatingSystem := instance.OperatingSystems().UnsortedList()[0]
+		zoneSet := sets.String{}
```

Comment:
I'm having a hard time understanding why this is so complicated to fake a zone. Can we make some helpers?

Reply:
It's possible that I'm just over-complicating things by mis-understanding how the fakes should work. I need the instance I fake create to match the requirements, right? So I first need to filter the offerings by what capacity types are allowed by the requirements and then by what zones are allowed by the requirements.

I _do_ think that I can pretty easily eliminate lines 58 and 59 by just picking the first offering that matches. And I think that lines 44-46 are unnecessary entirely now so I can cut those.

Not sure if I'm following your helper method ask, are you just wanting me to de-compose this code chunk so that it's shorter?